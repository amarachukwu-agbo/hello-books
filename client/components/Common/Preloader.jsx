import React from 'react';

/**
 * @description stateless component for loader
 * that is displayed while data is loading
 *
 * @returns {Node} - react node containing the Preloader component
 */
const Preloader = () => (
    <div className="preloader-wrapper medium active">
    <div className="spinner-layer spinner-blue-only">
      <div className="circle-clipper left">
        <div className="circle"></div>
      </div><div className="gap-patch">
        <div className="circle"></div>
      </div><div className="circle-clipper right">
        <div className="circle"></div>
      </div>
    </div>
  </div>
);

export default Preloader;
