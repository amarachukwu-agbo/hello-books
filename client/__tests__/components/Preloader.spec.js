import React from 'react';
import renderer from 'react-test-renderer';
import Preloader from '../../components/Common/Preloader';

describe('<Preloader>', () => {
  it('should render the preloader component correctly', () => {
    const tree = renderer.create(<Preloader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
