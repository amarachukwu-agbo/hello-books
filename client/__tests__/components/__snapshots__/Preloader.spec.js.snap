// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader> should render the preloader component correctly 1`] = `
<div
  className="preloader-wrapper medium active"
>
  <div
    className="spinner-layer spinner-blue-only"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
</div>
`;
